src/containers/Login/test/login.test.tsx

```diff
@@ -80,4 +89,55 @@ describe('Login', () => {
     expect(emailInput).toHaveValue('');
     expect(passwordInput).toHaveValue('');
   });
+
+  test('renders to the / route after successful login by employee', async () => {
+    authMockData();
+    renderLogin();
+
+    //send inputs to the form and submit
+    await formInputs('employee@employee.com', 'Password123!');
+  });
+
+  test('renders to the /onboarding route if employee logs in for the first time', async () => {
+    authMockData2();
+    renderLogin();
+
+    //send inputs to the form and submit
+    await formInputs('test@example.com', 'Password123!');
+  });
+});
+
+describe('Login Fails', () => {
+  test('error message is displayed when login fails and disapper after few seconds', async () => {
+    errorMockData();
+    renderLogin();
+    const submitButton = screen.getByRole('button', { name: /Log in/i });
+
+    // Submit the form
+    fireEvent.click(submitButton);
+
+    // Check if the error message is displayed
+    expect(screen.getByRole('alert')).toHaveClass('MuiAlert-standardError');
+    expect(
+      screen.getByText(/email or password is incorrect/i)
+    ).toBeInTheDocument();
+
+    // Wait for the error message to disappear and wrapping the transition inside a test in act(...).
+    await act(async () => {
```

Comment:
what does this code do?

Reply:
As we are waiting for the error message to disapper there is the change in state so need to wrap the waiting Promise inside the act function.